Sources/Build/BuildDelegate.swift

```diff
@@ -218,6 +217,12 @@ public final class BuildDelegate: BuildSystemDelegate {
         // https://forums.swift.org/t/allow-self-x-in-class-convenience-initializers/15924
         self.outputStream = outputStream as? ThreadSafeOutputByteStream ?? ThreadSafeOutputByteStream(outputStream)
         self.progressAnimation = progressAnimation
+
+        let buildConfig = plan.buildParameters.configuration.dirname
+        swiftParsers = Dictionary(uniqueKeysWithValues: plan.targetMap.compactMap({ (target, description) in
+            guard case .swift = description else { return nil }
+            return (target.getCommandName(config: buildConfig), SwiftCompilerOutputParser(delegate: self))
```

Comment:
We should capture the delegate weakly. We also need to change the delegate property to weak in `SwiftCompilerOutputParser`